### PR TITLE
`tagQuery()` docs; Add `.cssSelector` parameter to tag methods

### PR DESCRIPTION
**Pull request by cpsievert**
Closes https://github.com/rstudio/htmltools/issues/218
Closes https://github.com/rstudio/htmltools/issues/234
Closes https://github.com/rstudio/htmltools/issues/251

### CODE REVIEW

File: R/tag_query.R [hunk 1]
```diff
@@ -394,109 +394,14 @@ as.character.htmltools.tag.query <- function(x, ...) {
...
-#' z$asTags()
-#' #> <div></div>
-#' ```
-#'
-#'
-#' @section Limitations:
-#'
-#'   `tagQuery()`s can **not** be used directly within typical `tag` locations.
-#'   An error should be thrown. Instead, please call `$asTags()` to retrieve the
-#'   tag structures of the selected tag elements or root element respectively.
```
Comment (cpsievert) [inline R/tag_query.R:491]: These concepts won't go away completely, they'll just likely be towards the end of the tagQuery article

File: R/tag_query.R [hunk 2]
```diff
@@ -713,144 +580,142 @@ tagQuery_ <- function(
...
+
         #' ## Generic methods
         #'
-        #' * `$each(fn)`: Perform function `fn` on each of the selected
-        #' elements. `fn` should accept two arguments: a selected element and
+        #' * `$each(fn)`: Modify each selected tag with a function `fn`.
+        #' `fn` should accept two arguments: a selected element and
         #' the selected element's position within the selected elements. This
-        #' argument order is different than jQuery's `$().each()` as there is no
+        #' argument order mimics jQuery's `$().each()` as there is no
```
Comment (cpsievert) [inline R/tag_query.R:718]: I'll clean this up once I actually understand how `$each()` works. Is this expected?

```r
> (tagQuery(div(div(), div()))$find("div")$each(function(x, i) x <- a()))
Root:
<div>
  <div></div>
  <div></div>
</div>

Selected:
[[1]]
<div></div>
[[2]]
<div></div>
```

File: R/tag_query.R [hunk 3]
```diff
@@ -394,109 +394,15 @@ as.character.htmltools.tag.query <- function(x, ...) {
...
+#' @param tags A [tag()], [tagList()], or [list()] of tags. If a `list()` or
 #'   `tagList()` is provided, a `tagList()` will be returned when calling
 #'   `$asTags()`.
-#' @return A `tagQuery()` object. The `tag` supplied will be considered the
-#'   `root` object. At the time of initialization, the `root` is also considered
-#'   the single selected item. If any selections are made, the selected elements
-#'   will be updated.
+#' @return A list of methods for working with the "root" `tags` as well as
+#'   "selected" subset of `tags` (by default, `tags` is selected, but methods
+#'   such as `$find()` may be used to update the selection).
```
Comment (cpsievert) [inline R/tag_query.R:405]: Change this to mention `$root()` and `$selected()`

File: NEWS.md [hunk 4]
```diff
@@ -2,7 +2,9 @@
 
 ## New Features & Improvements
 
-* Added `tagQuery(tags)`. A tag query object implements many popular features of jQuery.  Similar to jQuery, tag query objects can find internal html using CSS selections. Given a selection (which defaults to the original `tags`), many alterations may be performed before converting the tag query object back to tag objects. (#208)
+* `{htmltools}` now has it's own `{pkgdown}` site hosted at <https://rstudio.github.io/htmltools/>.
```
Comment (wch) [inline NEWS.md:5]:
```suggestion
* `{htmltools}` now has its own `{pkgdown}` site hosted at <https://rstudio.github.io/htmltools/>.
```

**Comment by wch** [comment]
We should make sure the vignette is only a part of the website, and not included in the built package, because it can bloat the package size.

**Review by wch — APPROVED** [review]

File: NEWS.md [hunk 5]
```diff
@@ -2,7 +2,9 @@
...
+
+* The new `tagQuery()` function provides a [jQuery](https://jquery.com/) inspired interface to query and/or modify HTML `tag()` (e.g., `div()`) or `tagList()` objects. To learn more, see the [{pkgdown} article](). (#208)
```
Comment (cpsievert) [inline NEWS.md:7]: TODO: fill in the link

File: R/tag_query.R [hunk 6]
```diff
@@ -343,8 +343,7 @@ shinyTagEnvStr <- "<!-- htmltools.tag.env -->"
 
 #' @export
 as.tags.htmltools.tag.env <- function(x, ...) {
-  stop("Method not allowed", call. = TRUE)
-  # as.tags(tagEnvToTags(x), ...)
+  as.tags(tagEnvToTags(x), ...)
 }
```
Comment (cpsievert) [inline R/tag_query.R:347]: Revert and maybe provide a better message

File: R/tag_query.R [hunk 7]
```diff
@@ -597,280 +501,244 @@ tagQuery_ <- function(
...
           )
         },
-        #' * `$reset()`: A new `tagQuery()` object will be created with the
-        #' selected items set to the top level tag objects.
-        reset = function() {
+        #' ### Reset
+        #'
+        #' * `$resetSelection()`: Reset selected tags to the `$root()` tag. Useful
+        #' in combination with `$replaceWith()` since it empties the selection.
+        resetSelection = function() {
```
Comment (cpsievert) [inline R/tag_query.R:597]: Undo this change and rebase off of #235 when merged.

Also, update vignette to use this new name

File: R/tag_query.R [hunk 8]
```diff
@@ -882,23 +750,21 @@ tagQuery_ <- function(
...
           rebuild_()
           # Allows `$print()` to know if there is a root el
           tagQueryPrint(root, selected_)
           invisible(self)
         }
+        #' @examples
+        #'
+        #' tagQ <- tagQuery(div(a()))
+        #' tagQ$find("a")$addClass("foo")
+        #' tagQ
```
Comment (cpsievert) [inline R/tag_query.R:767]:
```suggestion
        #' tagQ
        #'
        #' # To learn more, visit https://rstudio.github.io/htmltools/articles/tagQuery.html
```

**Review by schloerke — COMMENTED** [review]
Your work LGTM!

Diff outputs in the performance section would be much easier given #249

**Comment by cpsievert** [comment]
2c95a28 closes https://github.com/rstudio/htmltools/issues/234
